### ISSUE TITLE
Sometimes the `follow_current_file` option doesn't work

### ISSUE THREAD

**Issue by nyngwang** [nvim-neo-tree/neo-tree.nvim#109]
As title. I don't know how to reproduce this. But I have experienced this bug many times recently. When this happen the cursor will always stay at the top/root. To resolve this, I will have to close `nvim` process and run it again.

---

### debug info.

```lua
      require('neo-tree').setup {
        open_files_in_last_window = true, -- move cursor correctly on close.
        enable_git_status = true,
        filesystem = {
          follow_current_file = true,
          filters = {
            show_hidden = true,
            respect_gitignore = false
          },
          -- many more
        }
      }
```

### NVIM

```
NVIM v0.7.0-dev+994-g5be2cdd91
Build type: Release
LuaJIT 2.1.0-beta3
```

**Comment by cseickel** [comment]
I'll keep my eye out for this. Let me know if you notice any patterns that can help me pinpoint the issue.

**Comment by nyngwang** [comment]
@cseickel: I think this happened when I change the current working directory.

**Comment by cseickel** [comment]
If you have `bind_to_cwd = true` and `follow_current_file = true` AND you change the cwd on every BufferEnter, that's definitely a recipe for unpredictable behavior!

I think it should be able to handle it though. At least it worked with vim-rooter last I checked. Can you link to your plugin so I can test it out?

**Comment by nyngwang** [comment]
> Can you link to your plugin so I can test it out?

https://github.com/nyngwang/NeoRoot.lua

It change the cwd for each file without any settings.

**Comment by cseickel** [comment]
I don't think it has anything to do with the plugin, it seems like that is a window local cwd which neo-tree does not respond to. I would be the root problem is also the cause of #111 ... I'll review this part of the code and look for issues.

**Comment by nyngwang** [comment]
@cseickel Definitely worth doing so, if this stop happening in I will also update to this thread~

**Comment by cseickel** [comment]
@nyngwang The latest commit on **main** might fix this issue. Can you try it out and let me know if you think it is fixed?

**Comment by nyngwang** [comment]
Absolutely yes :) If this might still happen I will tag you.

**Comment by cseickel** [comment]
@nyngwang It's possible the fix from #105 will fix this problem as well. Let me know.

**Comment by nyngwang** [comment]
@cseickel: I think it's resolved. These are the steps I used to test it:

1. Open 4 splits, all files are under different file paths far, farrrrr away.
2. Randomly choose one split, and change its buffer to `plugins.lua` where I trigger packer.nvim
3. After the update of all plugins by packer.nvim, open neo-tree via `NeoTreeToggle` and check whether the cursor behaves correctly.

It succeeded in the first run. I think we can close it now :)
